wpilibc/src/test/native/cpp/TracerTest.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <string>
+#include <string_view>
+#include <thread>
+
+#include <gtest/gtest.h>
+#include <networktables/DoubleTopic.h>
+#include <networktables/NetworkTableInstance.h>
+
+#include "frc/Tracer.h"
+#include "frc/simulation/SimHooks.h"
+
+TEST(TracerTest, TraceFunc) {
+  frc::Tracer::ResetForTest();
+  frc::Tracer::SetThreadName("TracerTest");
+
+  frc::sim::PauseTiming();
+
+  frc::Tracer::StartTrace("Test1");
+  frc::Tracer::TraceFunc("Test2", []() { frc::sim::StepTiming(400_ms); });
+  frc::sim::StepTiming(100_ms);
+  frc::Tracer::EndTrace();
+
+  auto test1Entry = nt::NetworkTableInstance::GetDefault().GetEntry(
+      "/Tracer/TracerTest/Test1");
+  auto test2Entry = nt::NetworkTableInstance::GetDefault().GetEntry(
+      "/Tracer/TracerTest/Test1/Test2");
+
+  EXPECT_TRUE(test1Entry.GetDouble(0.0) - 500.0 < 1.0);
+  EXPECT_TRUE(test2Entry.GetDouble(0.0) - 400.0 < 1.0);
```

Comment:
If the default value is zero, the value will be negative and will satisfy the condition. If you need some tolerance, use `EXPECT_NEAR`, or just `EXPECT_EQ` if you want to be exact.